bootstrap/util/configowner_test.go

```diff
@@ -20,113 +20,134 @@ import (
+			},
+			Status: clusterv1.MachinePoolStatus{
+				InfrastructureReady: true,
+			},
+		}
 
-	c := fake.NewFakeClientWithScheme(scheme)
-	objm := metav1.ObjectMeta{
-		OwnerReferences: []metav1.OwnerReference{
-			{
-				Kind:       "Machine",
-				APIVersion: clusterv1.GroupVersion.String(),
-				Name:       "my-machine",
+		c := fake.NewFakeClientWithScheme(scheme, myPool)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "MachinePool",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       "my-machine-pool",
+					},
+				},
+				Namespace: "my-ns",
+				Name:      "my-resource-owned-by-machine-pool",
 			},
-		},
-		Namespace: "my-ns",
-		Name:      "my-resource-owned-by-machine",
-	}
-	_, err := GetConfigOwner(context.TODO(), c, objm)
-	if err == nil {
-		t.Fatal("expected error, got nil")
-	}
-}
+		}
+		configOwner, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(configOwner).ToNot(BeNil())
+		g.Expect(configOwner.ClusterName()).To(BeEquivalentTo("my-cluster"))
+		g.Expect(configOwner.IsInfrastructureReady()).To(BeTrue())
+		g.Expect(configOwner.IsControlPlaneMachine()).To(BeFalse())
+		g.Expect(configOwner.DataSecretName()).To(BeNil())
+	})
 
-func TestGetConfigOwnerNoOwner(t *testing.T) {
-	scheme := runtime.NewScheme()
-	if err := clusterv1.AddToScheme(scheme); err != nil {
-		t.Fatal("failed to register cluster api objects to scheme")
-	}
+	t.Run("return an error when not found", func(t *testing.T) {
+		c := fake.NewFakeClientWithScheme(scheme)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Kind:       "Machine",
+						APIVersion: clusterv1.GroupVersion.String(),
+						Name:       "my-machine",
+					},
+				},
+				Namespace: "my-ns",
+				Name:      "my-resource-owned-by-machine",
+			},
+		}
+		_, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(err).To(HaveOccurred())
+	})
 
-	c := fake.NewFakeClientWithScheme(scheme)
-	objm := metav1.ObjectMeta{
-		OwnerReferences: []metav1.OwnerReference{},
-		Namespace:       "my-ns",
-		Name:            "my-resource-owned-by-machine",
-	}
-	configOwner, err := GetConfigOwner(context.TODO(), c, objm)
-	if err != nil {
-		t.Fatalf("did not expect an error but found one: %v", err)
-	}
-	if configOwner != nil {
-		t.Fatalf("expected nil, but got %v", configOwner)
-	}
+	t.Run("return nothing when there is no owner", func(t *testing.T) {
+		c := fake.NewFakeClientWithScheme(scheme)
+		obj := &bootstrapv1.KubeadmConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				OwnerReferences: []metav1.OwnerReference{},
+				Namespace:       "my-ns",
+				Name:            "my-resource-owned-by-machine",
+			},
+		}
+		configOwner, err := GetConfigOwner(context.TODO(), c, obj)
+		g.Expect(configOwner).To(BeNil())
+		g.Expect(err).To(BeNil())
```

Comment:
```suggestion
		g.Expect(err).NotTo(HaveOccurred())
```
Might also be better to have the assertion for `err` before the assertion for `configOwner`.